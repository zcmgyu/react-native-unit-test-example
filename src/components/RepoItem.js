import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Text, TouchableHighlight, View } from 'react-native';

const RepoItem = props => {
  const { description, id, name, stargazers_count } = props.repo;
  const itemStyle =
    (props.isSelected && [styles.item, styles.selected]) || styles.item;
  return (
    <TouchableHighlight
      onPress={() => {
        props.selectRepo ? props.selectRepo(id) : {};
      }}
      underlayColor="#E0F2F1"
    >
      <View style={itemStyle}>
        <Text style={styles.title}>{name}</Text>
        <Text style={styles.stars}>{`${stargazers_count} stars`}</Text>
        {props.isSelected ? <Text>{description}</Text> : null}
      </View>
    </TouchableHighlight>
  );
};

RepoItem.propTypes = {
  isSelected: PropTypes.bool,
  repo: PropTypes.object.isRequired,
  selectRepo: PropTypes.func
};

RepoItem.defaultProps = {
  isSelected: false
};

export const styles = StyleSheet.create({
  item: {
    paddingVertical: 8,
    paddingHorizontal: 24
  },
  selected: {
    backgroundColor: '#B2DFDB'
  },
  stars: {
    paddingBottom: 8
  },
  title: {
    fontSize: 16,
    fontWeight: 'bold',
    paddingBottom: 4
  }
});

export default RepoItem;
