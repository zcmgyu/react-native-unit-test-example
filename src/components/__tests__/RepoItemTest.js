import React from 'react';
import RepoItem from '../RepoItem';
import mockedRepos from '../../../config/jest/dataMock';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders a RepoItem using Snapshots', () => {
  const wrapper = renderer.create(
    <RepoItem repo={mockedRepos[0]} selectRepo={jest.fn()} />
  );
  expect(wrapper).toMatchSnapshot();
});
