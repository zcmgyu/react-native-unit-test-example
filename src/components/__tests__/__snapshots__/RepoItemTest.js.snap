// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a RepoItem using Snapshots 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "paddingHorizontal": 24,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "paddingBottom": 4,
        }
      }
    >
      Test 1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "paddingBottom": 8,
        }
      }
    >
      50000 stars
    </Text>
  </View>
</View>
`;
